### PR TITLE
Feature/search form

### PR DESCRIPTION
**Pull request by krootkipl**

### CODE REVIEW

File: src/app/app.component.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+@Component({
+    selector: 'app-root',
+    styleUrls: [
+        './app.component.css'
+    ],
+    templateUrl: './app.component.html',
+})
+export class AppComponent {
+    title = 'Minimal NgModule';
```
Comment (lzadylakpgs) [inline src/app/app.component.ts]: do you use this variable? if yes put type, if no please remove

File: src/app/app.component.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+}
```
Comment (lzadylakpgs) [inline src/app/app.component.ts]: please add enter at the end

File: src/app/app.module.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { NgModule }      from '@angular/core';
+import { BrowserModule } from '@angular/platform-browser';
+import { AppComponent }  from './app.component';
+import  { NavbarComponent} from "./shared/navbar/navbar.component";
```
Comment (lzadylakpgs) [inline src/app/app.module.ts]: remove space between import and bracket, add spaces between brackets and component name import { Navbar.. } from,
please check all files

File: src/app/app.module.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+        BrowserModule,
+        SearchModule,
+    ],
+    declarations: [
+        AppComponent,
+        NavbarComponent,
+    ],
+    bootstrap:    [ AppComponent ]
+})
+export class AppModule { }
```
Comment (lzadylakpgs) [inline src/app/app.module.ts]: please check all files and and enter at the end

File: src/app/search/search.component.css [hunk 5]
```diff
@@ -0,0 +1,7 @@
+.my-search-form{
```
Comment (lzadylakpgs) [inline src/app/search/search.component.css]: space between class name and bracket

File: src/app/search/search.component.html [hunk 6]
```diff
@@ -0,0 +1,17 @@
+<div class="my-search-form">
+
+    <div class="form-group">
+        <div class="col-md-4">
+            <input [(ngModel)]="artistQuery" type="text" id="searchArtist" name="searchArtist" placeholder="Search Artist" class="form-control input-md">
```
Comment (lzadylakpgs) [inline src/app/search/search.component.html]: please remember that one line shouldn't have more like 140 characters, if you see it's to long try to break, ex:
<input [(ngModel)]="artistQuery" 
       type="text" 
       id="searchArtist" 
       name="searchArtist" 
       placeholder="Search Artist" 
       class="form-control input-md">

File: src/app/search/search.component.css [hunk 7]
```diff
@@ -0,0 +1,7 @@
...
+    margin: 10px;
+}
+
+.form-group{
```
Comment (lzadylakpgs) [inline src/app/search/search.component.css]: we will also use SASS files and SUIT CSS convention for styling / template files
https://suitcss.github.io/

File: src/app/search/search.component.ts [hunk 8]
```diff
@@ -0,0 +1,24 @@
+import { Component } from '@angular/core';
+import {SearchService} from "./service/search.service";
```
Comment (lzadylakpgs) [inline src/app/search/search.component.ts]: spaces :)

File: src/app/search/search.component.ts [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+
+@Component({
+    selector: 'app-search',
+    templateUrl: './search.component.html',
+    styleUrls: ['./search.component.css'],
+    providers: [
+        SearchService,
+    ]
+
```
Comment (lzadylakpgs) [inline src/app/search/search.component.ts]: enter to remove

File: src/app/search/search.component.ts [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+})
+export class SearchComponent {
+    constructor(private _searchService: SearchService){
+    }
+
+    artistQuery: string;
```
Comment (lzadylakpgs) [inline src/app/search/search.component.ts]: move variables to top, above constructor

File: src/app/search/search.component.ts [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+    displayArtist: any;
+
+    searchClick(){
```
Comment (lzadylakpgs) [inline src/app/search/search.component.ts]: space between method and bracket, if method not return anything you should put void type after method name
searchClick(): void
and what do u think if we change the name to clickSearch() ?

File: src/app/search/search.module.ts [hunk 12]
```diff
@@ -0,0 +1,18 @@
+import { NgModule }      from '@angular/core';
+import  { SearchComponent } from "./search.component";
```
Comment (lzadylakpgs) [inline src/app/search/search.module.ts]: please clean up this file, and try to import angular files first and next components or other files like services

File: src/app/search/service/search.service.ts [hunk 13]
```diff
@@ -0,0 +1,35 @@
+import { Injectable } from '@angular/core';
+import {Http, Headers, Response} from "@angular/http";
+
+@Injectable()
+
```
Comment (lzadylakpgs) [inline src/app/search/service/search.service.ts]: enter to remove

File: src/app/search/service/search.service.ts [hunk 14]
```diff
@@ -0,0 +1,35 @@
...
+export class SearchService {
+
+    constructor(private http: Http){
+    }
+
+    private searchUrl = "https://api.spotify.com/v1/search";
```
Comment (lzadylakpgs) [inline src/app/search/service/search.service.ts]: try to create config file and move searchUlr and secretKey there

File: src/app/search/service/search.service.ts [hunk 15]
```diff
@@ -0,0 +1,35 @@
...
+    createAuthHeader(headers: Headers){
+        headers.append( "Accept", "application/json");
+        headers.append("Authorization", "Bearer " + this.secretKey);
+    }
+
+    getArtist(artistQuery = "Linkin Park"){
+        artistQuery = artistQuery.split(" ").join("+")
+        let headers = new Headers();
+        this.createAuthHeader(headers);
+        return this.http.get(this.searchUrl + "?q=" + artistQuery + "&type=artist", {headers})
```
Comment (lzadylakpgs) [inline src/app/search/service/search.service.ts]: try to use es6 template strings

File: src/app/search/service/search.service.ts [hunk 16]
```diff
@@ -0,0 +1,35 @@
...
+            .map((res:Response) => res.json());
+    }
+    searchArtist(artist: string){
+        this.getArtist(artist).subscribe(data => this.response = data);
+        console.log(this.response);
```
Comment (lzadylakpgs) [inline src/app/search/service/search.service.ts]: console log to remove

**Comment by krootkipl** [comment]
All requested changes done.

File: src/app/search/service/search.service.ts [hunk 17]
```diff
@@ -0,0 +1,34 @@
...
+    createAuthHeader(headers: Headers) {
+        headers.append( "Accept", "application/json");
+        headers.append("Authorization", "Bearer " + this.secretKey);
+    }
+
+    getArtist(artistQuery: string = "Linkin Park") {
+        artistQuery = artistQuery.split(" ").join("+")
+        let headers = new Headers();
+        this.createAuthHeader(headers);
+        return this.http.get(this.searchUrl + "?q=" + artistQuery + "&type=artist", {headers})
```
Comment (lzadylakpgs) [inline src/app/search/service/search.service.ts]: please use es6 template string

File: src/app/search/service/search.service.ts [hunk 18]
```diff
@@ -0,0 +1,34 @@
...
+export class SearchService {
+
+    constructor(private http: Http){
+    }
+
+    private searchUrl = "https://api.spotify.com/v1/search";
+
+    private secretKey = "BQDns0spirLV9h9EXU_V-R0AkCJVibifaT5rpyGexwC34FahBa4CbUNmX5ROLrqD84etYeAx_B_7YlNdt3pWKek30DjwUFXBoBzq-SSG3AN0dl7cl-au_lE1HrytILyfPz8grtvQX2VFqFw";
+
+    response: any;
```
Comment (lzadylakpgs) [inline src/app/search/service/search.service.ts]: this response have any type, could you ad interface for this response?

File: src/app/search/element/search-element.component.html [hunk 19]
```diff
@@ -1,15 +1,9 @@
...
-    <div class="artist-img-wrapper"
-         *ngFor="let entry of artist.images; let i=index">
-        <img *ngIf="i<1" src="{{entry.url}}" style="width: 200px; height: auto">
-    </div>
-    <div class="icon-wrapper">
-        <a target="_blank" href="{{artist.external_urls.spotify}}">
-            <img src="../../../assets/spotify_4096_black.png" style="width: 100px; height: auto">
-        </a>
+    <div (click)="playButton($event)" class="icon-wrapper">
+        <img src="../../../assets/spotify_4096_black.png" style="width: 100px; height: auto">
```
Comment (lzadylakpgs) [inline src/app/search/element/search-element.component.html]: don't use inline styling

File: src/app/search/element/search-element.component.html [hunk 20]
```diff
@@ -1,15 +1,9 @@
 <div class="artist-element">
```
Comment (lzadylakpgs) [inline src/app/search/element/search-element.component.html]: please use convention: https://github.com/suitcss/suit/blob/master/doc/naming-conventions.md so should by ArtistElemnt class name

File: src/app/shared/embedded-player/embedded-player.component.ts [hunk 21]
```diff
@@ -0,0 +1,34 @@
+import { Component, OnInit } from '@angular/core';
+import { PlayerService } from './services/player.service';
+import { DomSanitizer } from '@angular/platform-browser';
+
+
+@Component({
+    selector: 'app-embedded-player',
+    templateUrl: './embedded-player.component.html',
+    styleUrls: ['./embedded-player.component.css'],
+
```
Comment (lzadylakpgs) [inline src/app/shared/embedded-player/embedded-player.component.ts]: enter to remove

File: src/app/shared/embedded-player/embedded-player.component.ts [hunk 22]
```diff
@@ -0,0 +1,34 @@
...
+})
+export class EmbeddedPlayerComponent implements OnInit{
+
+    constructor(private _playerService: PlayerService, public sanitazer: DomSanitizer){
+
```
Comment (lzadylakpgs) [inline src/app/shared/embedded-player/embedded-player.component.ts]: enter to remove

File: src/app/shared/embedded-player/embedded-player.component.ts [hunk 23]
```diff
@@ -0,0 +1,34 @@
...
+    }
+
+    playUri: any = `https://open.spotify.com/embed?uri=spotify:user:1195887370:playlist:5k4JeojiXl6fcn3lLtQ6Vd`;
```
Comment (lzadylakpgs) [inline src/app/shared/embedded-player/embedded-player.component.ts]: it's string type, not any, and why you don't use yours globals.ts config file?

File: src/app/shared/embedded-player/embedded-player.component.ts [hunk 24]
```diff
@@ -0,0 +1,34 @@
...
+
+    ngOnInit() {
+        this._playerService.change.subscribe((uri:string) => {
```
Comment (lzadylakpgs) [inline src/app/shared/embedded-player/embedded-player.component.ts]: if you subscribe, you should also unsubscribe in ngOnDestroy

File: src/app/shared/embedded-player/embedded-player.component.ts [hunk 25]
```diff
@@ -0,0 +1,34 @@
...
+            this.selectedUri(uri);
+        })
+    }
+
+    selectedUri(uri: any){
+        this.playUri = "https://open.spotify.com/embed?uri=" + uri;
```
Comment (lzadylakpgs) [inline src/app/shared/embedded-player/embedded-player.component.ts]: the same like above, move url to const, and use es6 https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Template_literals

File: src/app/shared/navbar/navbar.component.html [hunk 26]
```diff
@@ -1,5 +1,13 @@
 <nav class="navbar-wrapper">
-    <a routerLink="/search" routerLinkActive="active">SEARCH</a>
-    <a routerLink="/playlists" routerLinkActive="active">PLAYLISTS</a>
-    <a routerLink="/user" routerLinkActive="active">USER</a>
+    <div class="navbar-inner-wrapper">
+        <div id="navbar-links">
+            <h3>SPOTIFY APP</h3>
+            <a routerLink="/search" routerLinkActive="active">SEARCH</a>
```
Comment (lzadylakpgs) [inline src/app/shared/navbar/navbar.component.html]: You should use styles to make capitalize strings in template

**Comment by lzadylakpgs** [comment]
This pull request it's to big, we should create new one, from this feature

File: src/app/playlists/playlist-element/playlist-element.component.html [hunk 27]
```diff
@@ -1,11 +1,11 @@
-<div class="playlist-element">
-    <div class="playlist-element-wrapper">
-        <p>${{playlist.name}}</p>
-        <div *ngFor="let image of playlist.images; let i=index">
-            <img *ngIf="i<1" src="{{image.url}}" style="width: 200px; height: auto">
+<div class="PlaylistElement">
+    <div class="PlaylistElement-wrapper">
+        <p>{{ playlist.name }}</p>
+        <div *ngIf="playlist.images.length">
+            <img src="{{ playlist.images[0].url }}" class="PlaylistElement-image">
```
Comment (lzadylakpgs) [inline src/app/playlists/playlist-element/playlist-element.component.html]: +1

File: src/app/search/search-element/search-element.component.html [hunk 28]
```diff
@@ -1,12 +1,12 @@
-<div class="artist-element">
-    <div class="artist-info-wrapper">
+<div class="SearchElement">
+    <div class="SearchElement-infoWrapper">
         <p>{{artist.name}}</p>
         <p>{{artist.popularity}}</p>
-        <div *ngFor="let image of artist.images; let i=index">
-            <img *ngIf="i<1" src="{{image.url}}" style="width: 200px; height: auto">
+        <div *ngIf="artist.images.length">
+            <img src="{{ artist.images[0].url }}" class="SearchElement-image">
```
Comment (lzadylakpgs) [inline src/app/search/search-element/search-element.component.html]: +1

File: src/app/search/search-element/search-element.component.html [hunk 29]
```diff
@@ -1,12 +1,12 @@
-<div class="artist-element">
-    <div class="artist-info-wrapper">
+<div class="SearchElement">
+    <div class="SearchElement-infoWrapper">
         <p>{{artist.name}}</p>
```
Comment (lzadylakpgs) [inline src/app/search/search-element/search-element.component.html]: brackets {{ NAME }}

File: src/app/search/search.component.scss [hunk 30]
```diff
@@ -0,0 +1,11 @@
+.SearchComponent {
+    width: 1200px;
+    margin: 0 auto;
+    color: white;
+
+    .SearchComponent-wrapper {
```
Comment (lzadylakpgs) [inline src/app/search/search.component.scss]: $-wrapper

File: src/app/shared/navbar/navbar.component.scss [hunk 31]
```diff
@@ -0,0 +1,39 @@
...
+    }
+}
+
+#navbar-links {
+    float: left;
+
+    a {
+        text-decoration: none;
+        border-radius: 4px;
+        background-color: aliceblue;
```
Comment (lzadylakpgs) [inline src/app/shared/navbar/navbar.component.scss]: once you use colors like: white / aliceblue / in other cases  color: #039be5; background-color: #CFD8DC; it's not good practice, keep one style

File: src/app/shared/navbar/navbar.component.scss [hunk 32]
```diff
@@ -0,0 +1,39 @@
+.NavbarComponent {
```
Comment (lzadylakpgs) [inline src/app/shared/navbar/navbar.component.scss]: +1

File: src/app/shared/navbar/navbar.component.scss [hunk 33]
```diff
@@ -0,0 +1,39 @@
...
+    margin-bottom: 10px;
+
+    &-wrapper {
+        margin: 0 auto;
+        padding: 5px;
+        width: 1200px;
+    }
+}
+
+#navbar-links {
```
Comment (lzadylakpgs) [inline src/app/shared/navbar/navbar.component.scss]: I think you should change this id to class, and keep inside .NavbarComponent
for example .NavbarComponent-links

File: src/app/shared/navbar/navbar.component.scss [hunk 34]
```diff
@@ -0,0 +1,39 @@
...
+            background-color: #CFD8DC;
+        }
+
+        &.active {
+            color: #039be5;
+        }
+    }
+}
+
+#navbar-player {
```
Comment (lzadylakpgs) [inline src/app/shared/navbar/navbar.component.scss]: the same like above, .NavbarComponent-player

File: src/app/playlists/playlist-element/playlist-element.component.html [hunk 35]
```diff
@@ -1,11 +1,11 @@
-<div class="PlaylistElement col-md-12">
-    <div class="PlaylistElement-wrapper col-md-4">
+<div class="PlaylistElement panel">
```
Comment (lzadylakpgs) [inline src/app/playlists/playlist-element/playlist-element.component.html]: +1

Comment (lzadylakpgs) [reply src/app/playlists/playlist-element/playlist-element.component.html]: panel have class header, body and footer

File: src/app/search/display/search-display.component.scss [hunk 36]
```diff
@@ -0,0 +1,6 @@
+.SearchDisplay {
+
+  &-column {
+    display: block;
```
Comment (lzadylakpgs) [inline src/app/search/display/search-display.component.scss]: display block is default for div

File: src/app/search/display/search-display.component.scss [hunk 37]
```diff
@@ -0,0 +1,6 @@
...
+  }
+}
```
Comment (lzadylakpgs) [inline src/app/search/display/search-display.component.scss]: new line

File: src/app/search/search-elements/artist-element/artist-element.component.html [hunk 38]
```diff
@@ -0,0 +1,8 @@
+<div class="SearchElement panel">
+    <div class="SearchElement-infoWrapper">
+        <p>Artist: {{ artist.name }}</p>
+    </div>
+    <div *ngIf="artist.images.length" (click)="playButton($event)">
+        <img src="{{ artist.images[0].url }}" class="SearchElement-image img-responsive">
+    </div>
+</div>
```
Comment (lzadylakpgs) [inline src/app/search/search-elements/artist-element/artist-element.component.html]: new line

File: src/app/search/search-elements/track-element/track-element.component.html [hunk 39]
```diff
@@ -0,0 +1,8 @@
+<div class="TrackElement panel">
+    <div class="TrackElement-infoWrapper">
+        <p>Track: {{ track.name }}</p>
+    </div>
+    <div *ngIf="track.album.images.length" (click)="playButton($event)">
+        <img src="{{ track.album.images[0].url }}" class="TrackElement-image img-responsive">
+    </div>
+</div>
```
Comment (lzadylakpgs) [inline src/app/search/search-elements/track-element/track-element.component.html]: new line

File: src/app/search/search.component.scss [hunk 40]
```diff
@@ -2,6 +2,7 @@
 
 .SearchComponent {
     color: white;
+    display: block;
```
Comment (lzadylakpgs) [inline src/app/search/search.component.scss]: it's not default block?

**Pull request merged**